lib/fastly.rb

```diff
@@ -46,6 +46,10 @@ class Fastly
   #
   # Some methods require full username and password rather than just auth token.
   def initialize(opts)
+    if (opts[:password] && opts[:user].nil?)
```

Comment:
Sorry for being unclear on the previous PR. I think we should validate all of the configuration options that come in. That is, we should make sure that either `key` is passed or both `user` and `password` are both passed. This condition is specific to the case of someone accidentally sending `password` and not `user`, but doesn't solve the scenario where someone misspells an option, for example.